### PR TITLE
[GPU Plugin] Fix the same cache hashID generated with different property setting

### PR DESCRIPTION
**Pull request by yangwang201911**
update properties list that affect the compilation process of GPU.
### Details:
 - Append property `ov::hint::performance_mode` 
 - Append property `ov::hint::dynamic_quantization_group_size`

### Tickets:
 - 150566

### CODE REVIEW

**Review by e-ddykim — APPROVED** [review]

**Review by zhaixuejun1993 — APPROVED** [review]
LGTM

**Review by isanghao — COMMENTED** [review]
Discussed offline with Peter: could you add dynamic quantization group size here as well?

**Comment by yangwang201911** [comment]
> Discussed offline with Peter: could you add dynamic quantization group size here as well?

Updated. @isanghao

File: src/plugins/intel_gpu/src/plugin/plugin.cpp [hunk 1]
```diff
@@ -510,11 +510,12 @@ ov::Any Plugin::get_metric(const std::string& name, const ov::AnyMap& options) c
 }
 
 std::vector<ov::PropertyName> Plugin::get_caching_properties() const {
-    static const std::vector<ov::PropertyName> caching_properties =  {
+    static const std::vector<ov::PropertyName> caching_properties = {
         ov::PropertyName{ov::device::architecture.name(), PropertyMutability::RO},
         ov::PropertyName{ov::intel_gpu::execution_units_count.name(), PropertyMutability::RO},
         ov::PropertyName{ov::hint::inference_precision.name(), PropertyMutability::RW},
         ov::PropertyName{ov::hint::execution_mode.name(), PropertyMutability::RW},
+        ov::PropertyName{ov::hint::performance_mode.name(), PropertyMutability::RW},
```
Comment (peterchen-intel) [inline src/plugins/intel_gpu/src/plugin/plugin.cpp:518]:
```suggestion
        ov::PropertyName{ov::hint::performance_mode.name(), PropertyMutability::RW},
        ov::PropertyName{ov::hint::dynamic_quantization_group_size.name(), PropertyMutability::RW},
```

Comment (peterchen-intel) [reply src/plugins/intel_gpu/src/plugin/plugin.cpp:518]: Add dynamic_quantization_group_size as well.

File: src/plugins/intel_gpu/src/plugin/plugin.cpp [hunk 2]
```diff
@@ -510,11 +510,14 @@ ov::Any Plugin::get_metric(const std::string& name, const ov::AnyMap& options) c
...
+        ov::PropertyName{ov::hint::dynamic_quantization_group_size.name(), PropertyMutability::RW},
+        ov::PropertyName{ov::hint::dynamic_quantization_group_size.name(), PropertyMutability::RW},
```
Comment (peterchen-intel) [inline src/plugins/intel_gpu/src/plugin/plugin.cpp:520]:
```suggestion
                ov::PropertyName{ov::hint::dynamic_quantization_group_size.name(), PropertyMutability::RW},
```

Comment (peterchen-intel) [inline src/plugins/intel_gpu/src/plugin/plugin.cpp:520]:
```suggestion
```

**Review by isanghao — APPROVED** [review]

**Pull request merged**